msprime/ancestry.py

```diff
@@ -1510,7 +1510,7 @@ def _run_until(self, end_time, event_chunk=None, debug_func=None):
                     "Please open an issue on GitHub"
                 )
             logger.debug(
-                "time=%g ancestors=%d ret=%s", self.time, self.num_ancestors, ret
+                f"time={self.time} ancestors={self.num_ancestors} ret={repr(ret)}"
```

Comment:
Why this change? Are these logging converters not supported after 3.11? They only generate the string as needed with log level, so are useful

Reply:
In Py3.11 `IntEnum`'s  `__str__` method has reverted to that of a normal int. The other way to fix this is to change the checked for text to test for an int instead of the string. I'll change it to that.